src/adapter/adapter.ts

```diff
@@ -15,6 +15,7 @@ interface AdapterEventMap {
     zdoResponse: [clusterId: Zdo.ClusterId, response: ZdoTypes.GenericZdoResponse];
     disconnected: [];
     deviceLeave: [payload: AdapterEvents.DeviceLeavePayload];
+    sourceRoute: [payload: any];
```

Comment:
Should this be typed too? When I tried I was getting typing errors in the emit line in the zstack adapter class, but I didn't get much deeper than that.

Reply:
The CI check would indicate that yes, I should fix this!

Reply:
Definitely needs typing, and definitely needs to be a generic form that works with all adapters (not the payload of zstack). Although, it's unlikely some adapters will even be able to support this.